modules/electrophysiology_browser/ajax/get_electrophysiology_session_data.php

```diff
@@ -0,0 +1,203 @@
+    IN ("bdf", "cnt", "edf", "set", "vhdr", "vsm") ORDER BY pf.SessionID';
+
+    $sessions                = $db->pselect($query, array());
+    $sessions                = array_column($sessions, 'SessionID');
+    $response['patient']     = getSubjectData($sessionID);
+    $response['database']    = array_values(getFilesData($sessionID));
+    $response['sessions']    = $sessions;
+    $currentIndex            = array_search($sessionID,$sessions);
+    $response['nextSession'] = isset($sessions[$currentIndex+1]) ?
+        $sessions[$currentIndex+1] : '';
+    $response['prevSession'] = isset($sessions[$currentIndex-1]) ?
+        $sessions[$currentIndex-1] : '';
+
+    return $response;
+}
+function getSubjectData($sessionID)
+{
+    $subjectData = array();
+    $timePoint   =& \TimePoint::singleton($sessionID);
+    $candidate   =& \Candidate::singleton($timePoint->getCandID());
+
+    $subjectData['pscid']       = $candidate->getPSCID();
+    $subjectData['dccid']       = $timePoint->getCandID();
+    $subjectData['visit_label'] = $timePoint->getVisitLabel();
+    $subjectData['sessionID']   = $sessionID;
+    $subjectData['site']        = $timePoint->getPSC();
+    $subjectData['dob']         = $candidate->getCandidateDoB();
+    $subjectData['sex']         = $candidate->getCandidateSex();
+    $subjectData['subproject']  = $timePoint->getData('SubprojectTitle');
+    $subjectData['output_type'] = $_REQUEST['outputType'];
+
+    return $subjectData;
+}
+function getFilesData($sessionID)
+{
+    $db = \Database::singleton();
+
+    $fileCollection = array();
+    $outputType     = $_REQUEST['outputType'];
+    $params['SID']  = $sessionID;
+    $query          = 'SELECT pf.PhysiologicalFileID, pf.FilePath FROM
+    physiological_file pf ';
+    //WHERE SessionID=:SID
+    if ($outputType != 'all_types') {
+        $query .= 'LEFT JOIN physiological_output_type pot ON ';
+        $query .= 'pf.PhysiologicalOutputTypeID=pot.PhysiologicalOutputTypeID ';
+        $query .= 'WHERE SessionID=:SID ';
+        if ($outputType == 'raw') {
+            $query .= 'AND pot.OutputTypeName = "raw"';
+        } else if ($outputType == 'derivatives') {
+            $query .= 'AND pot.OutputTypeName = "derivatives"';
+        }
+    } else {
+        $query .= "WHERE SessionID=:SID";
+    }
+    $physiologicalFiles = $db->pselect($query, $params);
+    $files = [];
+    foreach ($physiologicalFiles as $file) {
+        $fileSummary          = array();
+        $physiologicalFileID  = $file['PhysiologicalFileID'];
+        $physiologicalFile    = $file['FilePath'];
+        $physiologicalFileObj = new \BIDSFile($physiologicalFileID);
+        $fileName             = basename(
+            $physiologicalFileObj->getParameter('FilePath')
+        );
+
+        $fileSummary['name'] = $fileName;
+        $fileSummary['task']['frequency']['sampling'] = $physiologicalFileObj->getParameter('SamplingFrequency');
```

Comment:
This part makes me want to cry. 😿 PTSD from the statistics module.

Is there anyway to refactor this into a function?